### ISSUE TITLE
Request: Detailed export

### ISSUE THREAD

**Issue by Albin-Gedda** [kauffman12/EQLogParser#191]
Export to XML is the single only reason I am still stuck with GamParse. And I want to retire it!

Is there any chance you can add an option to export the detailed tables from whatever I filter out in Fight List?
Any format works but preferably something natively supported by Excel :) Like xlsx, xml or json.

Personally I use the GamParse detailed XML files to create a small dashboard per fight to post in our Discord.

GamParse xml is missing (imo) a table for PlayerName;PlayerClass containing all the names in the export. If it were present, creating my own dashboards from the export would mean no more dependencies of local lookup tables or reading Apps ini or settings.

**Comment by kauffman12** [comment]
I haven't looked at Gamparse XML before now. Do you use all 5 sheets it creates? What does the Dashboard show? It looks like the XML is all individual hits or heals or whatever. Do you turn all that back into totals like the DPS summary is doing in the parser?

**Comment by Albin-Gedda** [comment]
It is a great base for me to save event data in a structured format, and it lets me go back and compare some details between events a lot easier than reloading saved logs in 2 or 3 opened parsers. I import all the xml files into single tables where I can filter whatever I want from it.

Having all this data in an easy to access format has helped in looking at some individual progression, evaluating apps for the last weeks, filter out who uses wrong spells still 4 weeks after new expansions, helping members with detailed info regarding wtf player x is actually casting since he is doing 50% more dmg etc.

Regarding the sheets, no I do not use all 5. I use the DPS sheet and also filter out all the playernames from spell sheet when I do dashboards, to pick up anyone present in event that has a 0 dps, and is not present in the DPS sheet (yes that happens in other guilds 😉).
But the XML file could be better, For example the current file is repeating saved event name on all rows instead of target name.
It requires me to have an external lookup table for Player;Class.
It adds the Pet Owner in the Attacker field, it should have another column for PlayerName and if attacker is a pet have the Player in PlayerName column, and if it is a player, then yeah... PlayerName would be in 2 columns, but whatever.
Any person present at the event that do no damage is excluded from the DPS table, and I guess that is correct, but it still means they have to be added by filtering it against Spell table and add them at Timestamp 1 with dmg 0.
If a player:Class table was included in the export, I guess that would just be the base for all names in the chart.
However, what I do think is good, is that it shows event time in seconds of event, not in log row real time. But I guess an exported table could have both.

If an export contains enough data to create some simple dashboards without relying on local lookup tables, it would be awesome!

![SoD_Parse 2024-07-14 Bidil's_Great_Adventure](https://github.com/user-attachments/assets/e48109e6-05b9-40ed-9655-2f0297b28c0b)

**Comment by kauffman12** [comment]
So, Player DPS and Spells cast sheets with some changes.

1. Opponent column should be the NPC name and not include event name.
2. Add a column for Pet Owner. The player name goes there and the Attacker would just be the pet.
3. Add a Player Class column.

Should the combined/event name be set in the first row of each sheet maybe? Or don't you need that?

**Comment by Albin-Gedda** [comment]
First. Thanks for considering this!

I guess my current needs should not limit an export when there is more data available. Having more means new ideas will be possible in the future.
I bet some in the community can create some nice stuff if most of the info is available in an export.
To make the imports easy to use from as many external type programs as possible, Data tabs should not include anything other than a single table.

You already have a couple of very nice calculated tables, with summaries and breakdowns.
Including those, as tabs, would be nice. And save me from having to calculate everything again if just making a totals chart for example.
They are small compared to the total size of an detailed export.


LOOKUPTABLE
 -- If the DPS table is split into Attacker;Player, pet names is not needed here, but can still be nice to have for some future idea
Player -- List all players present in the current Fight list selection, if they have done >0 damage or >0 spell casts or tanked for >0 damage.
Class -- Add players class
Pet -- Players PetName

DPS DETAILED:
 -- Every single row from the log that is some kind of damage
LogTime -- DateTime from log. ISO 8601 format for best compability with external apps.
EventTime -- Calcultate Time from 1 to x
Target -- Mob name
Attacker -- PlayerName or PetName. 
Player -- PlayerName again. Makes it a lot easier to create simple calcualtions without lookups.
Type -- Crush|Slash|Bite|pierce etc
Damage -- Damage number
Modifier -- Critical|Lucky etc

SPELL
LogTime -- DateTime from log. ISO 8601 format for best compability with external apps.
EventTime -- Calcultate Time from 1 to x
Caster -- Name of caster
Spell -- Spell name
Direction -- Cast;Receive
Rank -- This would be a very nice bonus for some analytics, having the clean spell name in Spell, and rank in a separate column

EVENT INFO (just a 2 column table with these as key/value)
EventName -- Value from Custom Title
EventLength -- Time in seconds between first NPC hit and last NPC death, or however you calculate this already
EventDateTime -- DateTime from first affected row in the log
EventTotalDmg -- Total damage
Maybe some more rows, including the things already calculated in blue text above the summary tables


That is what I was dreaming about regarding this.
But maybe others want to use some tanking table or heals tables.
If you were to implement this kind of export, the structure is in place. Adding an extra tab with other detailed data is quite easy if the community ask for it.

Again. I am thrilled that you did not dismissed this :)

**Comment by kauffman12** [comment]
Most the data is easily available in a bunch of lists so it shouldn't be too bad to do a basic data dump to a spreadsheet. I don't know how complicated I want to get with it considering I could spend more time adding views/charts to the parser itself.

**Comment by Albin-Gedda** [comment]
Any export of raw data would be nice.
As a base for event dashboards and being able to save event data in a structured format.
That would be all I need to go ahead with my ideas, I would read the player;class;petname from App settings and sort it locally.
Same with any calculation I want.
No need for adding ID or name in the file. I can sort that with the filename.

I also assumed the summaries and breakdowns were already in a format that would be ready to call and insert in tabs.

My list above was more to mitigate some of the problems I had years back when this stuff was new to me. (And to not having to redo some already done calculations).
If some user is in the the first phase of looking into Excel or some other way to work with the data, it is nice having it all available in one file.

**Comment by Albin-Gedda** [comment]
Just struck me.
A big reason to include a PlayerName;PlayerClass tab in an export is Personas.
Having a static list on my local pc will assign wrong class to anyone raiding on a persona.

**Comment by kauffman12** [comment]
I'm still planning on doing this btw. Just other more important things keep coming up.

**Comment by Albin-Gedda** [comment]
Just checking in.
Right clicking a name in DPS Summary and selecting DPS Log.
Letting DPS Log include multiple Names instead of being grayed out when selecting > 1 name would mean I can uninstall GP :)
(Also all numerical values on CSV exports are saved as text, inside double quotations, please consider changing them to clean numerical values without included formatting)

**Comment by kauffman12** [comment]
Yeah it breaks for me when spaces are used in the number format. 

It is pretty simple to just output raw numbers so maybe I'll make it an option you can set. It is pretty convenient if you want commas as you don't need to manually format columns after pasting them. Something like 'Export Formatted CSV' that defaults to on under Options.
